bundle/EventListener/InternalContentViewRouteListener.php

```diff
@@ -0,0 +1,156 @@
+
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use eZ\Publish\Core\MVC\Symfony\Routing\Generator\UrlAliasGenerator;
+use eZ\Publish\Core\MVC\Symfony\Routing\UrlAliasRouter;
+use eZ\Publish\Core\MVC\Symfony\SiteAccess;
+use EzSystems\EzPlatformAdminUiBundle\EzPlatformAdminUiBundle;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Controller\ControllerReference;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\HttpKernel\Fragment\FragmentHandler;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Routing\RouterInterface;
+use function class_exists;
+use function in_array;
+
+class InternalContentViewRouteListener implements EventSubscriberInterface
+{
+    /**
+     * @var \eZ\Publish\Core\MVC\ConfigResolverInterface
+     */
+    private $configResolver;
+
+    /**
+     * @var \Symfony\Component\HttpKernel\Fragment\FragmentHandler
+     */
+    private $fragmentHandler;
+
+    /**
+     * @var \Symfony\Component\Routing\RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var array
+     */
+    private $siteaccessGroups;
+
+    public function __construct(
+        ConfigResolverInterface $configResolver,
+        FragmentHandler $fragmentHandler,
+        RouterInterface $router,
+        array $siteaccessGroups
+    ) {
+        $this->configResolver = $configResolver;
+        $this->fragmentHandler = $fragmentHandler;
+        $this->router = $router;
+        $this->siteaccessGroups = $siteaccessGroups;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::REQUEST => ['onKernelRequest', 0],
+        ];
+    }
+
+    public function onKernelRequest(RequestEvent $event): void
+    {
+        if (!$event->isMasterRequest()) {
+            return;
+        }
+
+        $request = $event->getRequest();
+
+        if (!$this->isInternalContentViewRoute($request)) {
+            return;
+        }
+
+        $siteaccess = $request->attributes->get('siteaccess');
+
+        if (!$siteaccess instanceof SiteAccess || $this->isAdminSiteaccess($siteaccess)) {
+            return;
+        }
+
+        if (!$this->configResolver->getParameter('ng_site_api.enable_internal_view_route')) {
+            throw new NotFoundHttpException();
+        }
+
+        $event->setResponse($this->getResponse($request));
+    }
+
+    private function getResponse(Request $request): Response
+    {
+        if ($this->configResolver->getParameter('ng_site_api.redirect_internal_view_route_to_url_alias')) {
+            return new RedirectResponse($this->generateUrlAlias($request));
+        }
+
+        return new Response($this->renderView($request));
```

Comment:
Instead of manually rendering the view, can't you just do nothing and let it fallback to eZ?

Reply:
Yes, otherwise it would fall back on the default eZ templates, which is not useful.